src/ngx_http_lua_bodyfilterby.c

```diff
@@ -140,6 +146,19 @@ ngx_http_lua_body_filter_by_chunk(lua_State *L, ngx_http_request_t *r,
 
     lua_settop(L, 0);
 
+    if (ctx->body_exited) {
+        /* send last content */
+        for (cl = in; cl->next; cl = cl->next) {
+            /* do nothing */
+        }
+        /* make ngx_http_write_filter_module happy */
+        cl->buf->last_buf = 1;
+
+        ctx->eof = 1;
+
+        return NGX_OK;
```

Comment:
@agentzh Sorry, I have a special condition to forget to describe: because we get the content from the nginx's upstream module, so you can only add some Lua operation in the `bodyfilter` phase. can we exit `ngx.ERROR` at bodyfilter and let client detects truncation? How can I do? thanks a lot